pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go

```diff
@@ -177,143 +224,248 @@ func (n *clusterManagerController) sync(ctx context.Context, controllerContext f
 		}
 	}
 
-	// ClusterManager is deleting, we remove its related resources on hub
+	// Get clients of the hub cluster and the management cluster
+	hubKubeConfig, err := helpers.GetHubKubeconfig(ctx, n.operatorKubeconfig, n.operatorKubeClient, clusterManagerName, clusterManagerMode)
+	if err != nil {
+		return err
+	}
+	hubClient, hubApiExtensionClient, hubApiRegistrationClient, err := n.generateHubClusterClients(hubKubeConfig)
+	if err != nil {
+		return err
+	}
+	managementClient := n.operatorKubeClient // We assume that operator is always running on the management cluster.
+
+	// If the ClusterManager is deleting, we remove its related resources on hub
 	if !clusterManager.DeletionTimestamp.IsZero() {
-		if err := n.cleanUp(ctx, controllerContext, config); err != nil {
+		if err := cleanUpHub(ctx, controllerContext, clusterManagerMode, hubClient, hubApiExtensionClient, hubApiRegistrationClient, config); err != nil {
+			return err
+		}
+		if err := cleanUpManagement(ctx, controllerContext, hubClient, config); err != nil {
 			return err
 		}
-		return n.removeClusterManagerFinalizer(ctx, clusterManager)
+		return removeClusterManagerFinalizer(ctx, n.clusterManagerClient, clusterManager)
 	}
 
-	// Apply static files(Which don't require CABundle)
 	var relatedResources []operatorapiv1.RelatedResourceMeta
+
+	// Apply resources on the hub cluster
+	hubAppliedErrs, err := applyHubResources(clusterManagerNamespace,
+		clusterManagerMode,
+		config,
+		&relatedResources,
+		hubClient, hubApiExtensionClient, hubApiRegistrationClient,
+		n.configMapLister, n.recorder)
+	if err != nil {
+		return err
+	}
+
+	// Apply resources on the management cluster
+	currentGenerations, managementAppliedErrs, err := applyManagementResources(
+		ctx,
+		clusterManager,
+		config,
+		&relatedResources,
+		hubClient, hubKubeConfig,
+		managementClient, n.recorder, n.ensureSAKubeconfigs,
+	)
+	if err != nil {
+		return err
+	}
+
+	// Update status
+	errs := append(hubAppliedErrs, managementAppliedErrs...)
+	conditions := &clusterManager.Status.Conditions
+	observedKlusterletGeneration := clusterManager.Status.ObservedGeneration
+	if len(errs) == 0 {
+		meta.SetStatusCondition(conditions, metav1.Condition{
+			Type:    clusterManagerApplied,
+			Status:  metav1.ConditionTrue,
+			Reason:  "ClusterManagerApplied",
+			Message: "Components of cluster manager is applied",
```

Comment:
Components of cluster manager are applied

Reply:
fixed.